src/common/utils/MetaKeyUtils.cpp

```diff
@@ -1152,9 +1152,38 @@ GraphSpaceID MetaKeyUtils::parseLocalIdSpace(folly::StringPiece rawData) {
   return *reinterpret_cast<const GraphSpaceID*>(rawData.data() + offset);
 }
 
-GraphSpaceID MetaKeyUtils::parseDiskPartsSpace(folly::StringPiece rawData) {
+/**
+ * diskPartsKey = kDiskPartsTable + serializeHostAddr(HostAddr) + SpaceId + diskPath
+ * kDiskPartsTable = "__disk_parts__"
+ * HostAddr : see serializeHostAddr(), HostAddr = len(host.ip) + host.ip + host.port
+ * SpaceId: int32_t
+ * diskPath: string like "/data/storage/1/nebula/part_1"
+ */
+
+HostAddr MetaKeyUtils::parseDiskPartsHost(const folly::StringPiece& rawData) {
   auto offset = kDiskPartsTable.size();
-  return *reinterpret_cast<const GraphSpaceID*>(rawData.data() + offset);
+  size_t hostIpLen = *reinterpret_cast<const size_t*>(rawData.begin() + offset);
+  std::string hostAddrStr;
+  hostAddrStr.reserve(sizeof(size_t) + hostIpLen + sizeof(Port));
+  hostAddrStr.append(rawData.begin() + offset, sizeof(size_t) + hostIpLen + sizeof(Port));
+  return deserializeHostAddr(hostAddrStr);
```

Comment:
I think 
```c++
std::string hostAddrStr(rawData.begin() + offset, rawData.size() - offset);
return deserializeHostAddr(hostAddrStr);
```
will be enough?
Because the parseDiskPartsHost do not need to know the format which only `deserializeHostAddr` should know.

Reply:
in this case, hostAddrStr is hostAddr + spaceId + path, deserializeHostAddr will check the length to make sure the host is valid, so there use sizeof(size_t) + hostIpLen + sizeof(Port) to get the real length of hostAddr.